File: ExtApp/ExtApp.Web/app/model/authority/Menu.js

Ext.define('App.model.authority.Menu', {
    extend: 'Ext.data.Model',

    idProperty: 'ID',
    fields: [
        'ID',
        {
            name: 'PID',
            mapping: function (value) {
                if (value.PMenu == null) {
                    return 0;
                }
                return value.PMenu.ID;
            }
        },
        {
            name: 'PName',
            mapping: function (value) {
                if (value.PMenu == null) {
                    return '顶级菜单';
                }
                return value.PMenu.Name;
            }
        },
        'Code',
        'Name',
        {
            name: 'UrlTypeID',
            mapping: function (value) {
                if (value.UrlType == null) {
                    return 0;
                }
                return value.UrlType.ID;
            }
        },
        {
            name: 'UrlTypeCode',
            mapping: function (value) {
                if (value.UrlType == null) {
                    return '';
                }
                return value.UrlType.Code;
            }
        },
        {
            name: 'UrlTypeName',
            mapping: function (value) {
                if (value.UrlType == null) {
                    return '';
                }
                return value.UrlType.Name;
            }
        },
        'Url',
        {
            name: 'OpenTypeID',
            mapping: function (value) {
                if (value.OpenType == null) {
                    return 0;
                }
                return value.OpenType.ID;
            }
        },
        {
            name: 'OpenTypeCode',
            mapping: function (value) {
                if (value.OpenType == null) {
                    return '';
                }
                return value.OpenType.Code;
            }
        },
        {
            name: 'OpenTypeName',
            mapping: function (value) {
                if (value.OpenType == null) {
                    return '';
                }
                return value.OpenType.Name;
            }
        },
        {
            name: 'IconTypeID',
            mapping: function (value) {
                if (value.IconType == null) {
                    return 0;
                }
                return value.IconType.ID;
            }
        },
        {
            name: 'IconTypeCode',
            mapping: function (value) {
                if (value.IconType == null) {
                    return '';
                }
                return value.IconType.Code;
            }
        },
        {
            name: 'IconTypeName',
            mapping: function (value) {
                if (value.IconType == null) {
                    return '';
                }
                return value.IconType.Name;
            }
        },
        'Icon',
        'Sort',
        'Status',
        'Comment'
    ]
});

File: ExtApp/ExtApp.Model/Authority/Menu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NHibernate.Mapping.Attributes;

namespace ExtApp.Model
{
    /// <summary>
    /// 系统菜单表
    /// </summary>
    [Class(Table = "AppMenu")]
    public class Menu
    {
        /// <summary>
        /// 菜单ID
        /// </summary>
        [Id(Name = "ID")]
        [Generator(Class = "identity")]
        public virtual int ID { get; set; }

        /// <summary>
        /// 父节点编码
        /// </summary>
        [ManyToOne(ClassType = typeof(Menu), Column = "PID", Lazy = Laziness.False)]
        public virtual Menu PMenu { get; set; }

        /// <summary>
        /// 编码
        /// </summary>
        [Property(NotNull = true)]
        public virtual string Code { get; set; }

        /// <summary>
        /// 名称
        /// </summary>
        [Property(NotNull = true)]
        public virtual string Name { get; set; }

        /// <summary>
        /// Url类型
        /// </summary>
        [ManyToOne(ClassType = typeof(DicItem), Column = "UrlType", Lazy = Laziness.False)]
        public virtual DicItem UrlType { get; set; }

        /// <summary>
        /// 模块地址或url
        /// </summary>
        [Property]
        public virtual string Url { get; set; }

        /// <summary>
        /// 打开方式
        /// </summary>
        [ManyToOne(ClassType = typeof(DicItem), Column = "OpenType", Lazy = Laziness.False)]
        public virtual DicItem OpenType { get; set; }

        /// <summary>
        /// 图标类型
        /// </summary>
        [ManyToOne(ClassType = typeof(DicItem), Column = "IconType", Lazy = Laziness.False)]
        public virtual DicItem IconType { get; set; }

        /// <summary>
        /// 图标
        /// </summary>
        [Property]
        public virtual string Icon { get; set; }

        /// <summary>
        /// 排序
        /// </summary>
        [Property]
        public virtual int? Sort { get; set; }

        /// <summary>
        /// 状态（1-启用，0-禁用）
        /// </summary>
        [Property]
        public virtual int? Status { get; set; }

        /// <summary>
        /// 备注
        /// </summary>
        [Property]
        public virtual string Comment { get; set; }
    }
}
